lib/semantics/type.h

```diff
@@ -105,40 +112,42 @@ class ParamValue {
 class IntrinsicTypeSpec {
 public:
   TypeCategory category() const { return category_; }
-  int kind() const { return kind_; }
+  KindExpr &kind() { return kind_; }
```

Comment:
Why not have a `set_kind` member function? That seems more consistent with other classes.

If `Scope::InstantiateIntrinsicType` is the only place this is needed, it might be possible to have a way to make a copy of an intrinsic type with a different kind. Then a mutator wouldn't be needed at all.

Reply:
Will change to a standard accessor/mutator API, and look into whether an additional constructor would obviate the need for the mutator.